dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/expand/CuringParamsServiceImpl.java

```diff
@@ -69,8 +70,13 @@ public class CuringParamsServiceImpl implements CuringParamsService {
     private ProjectParameterMapper projectParameterMapper;
 
     @Override
-    public String convertParameterPlaceholders(String val, Map<String, String> allParamMap) {
-        return ParameterUtils.convertParameterPlaceholders(val, allParamMap);
+    public String convertParameterPlaceholders(String val, Map<String, Property> allParamMap) {
+        Map<String, String> paramMap = allParamMap
+                .entrySet()
+                .stream()
+                .filter(entry -> nonNull(entry.getValue().getValue()))
```

Comment:
I am not clear if we need to filter the null here.

Reply:
seems we have some NPE in some buildint parameter